src/client/dfs/duns.c

```diff
@@ -576,44 +573,97 @@ duns_parse_attr(char *str, daos_size_t len, struct duns_attr_t *attr)
 		D_GOTO(err, rc = EINVAL);
 	}
 
+	/** da_puuid is deprecated, but in case users are still using it parse the uuid there */
 	rc = uuid_parse(t, attr->da_puuid);
 	if (rc) {
 		D_ERROR("Invalid DAOS xattr format: pool UUID cannot be parsed\n");
 		D_GOTO(err, rc = EINVAL);
 	}
+	snprintf(attr->da_pool, DAOS_PROP_LABEL_MAX_LEN + 1,  "%s", t);
 
 	t = strtok_r(NULL, "/", &saveptr);
 	if (t == NULL) {
 		D_ERROR("Invalid DAOS xattr format (%s).\n", str);
 		D_GOTO(err, rc = EINVAL);
 	}
+
+	/** da_cuuid is deprecated, but in case users are still using it parse the uuid there */
 	rc = uuid_parse(t, attr->da_cuuid);
 	if (rc) {
 		D_ERROR("Invalid DAOS xattr format: container UUID cannot be parsed\n");
 		D_GOTO(err, rc = EINVAL);
 	}
+	snprintf(attr->da_cont, DAOS_PROP_LABEL_MAX_LEN + 1,  "%s", t);
 
 	rc = 0;
 err:
 	D_FREE(local);
 	return rc;
 }
 
+static int
+create_cont(daos_handle_t poh, struct duns_attr_t *attrp)
+{
+	int rc;
+
+	if (attrp->da_type == DAOS_PROP_CO_LAYOUT_POSIX) {
+		dfs_attr_t dfs_attr = {};
+
+		/** TODO: set Lustre FID here. */
+		dfs_attr.da_id = 0;
+		dfs_attr.da_oclass_id = attrp->da_oclass_id;
+		dfs_attr.da_chunk_size = attrp->da_chunk_size;
+		dfs_attr.da_props = attrp->da_props;
+		if (!uuid_is_null(attrp->da_cuuid))
+			rc = dfs_cont_create(poh, attrp->da_cuuid, &dfs_attr, NULL, NULL);
+		else
+			rc = dfs_cont_create(poh, &attrp->da_cuuid, &dfs_attr, NULL, NULL);
+	} else {
+		daos_prop_t	*prop;
+		int		 nr = 1;
+
+		if (attrp->da_props != NULL)
+			nr = attrp->da_props->dpp_nr + 1;
+
+		prop = daos_prop_alloc(nr);
+		if (prop == NULL) {
+			D_ERROR("Failed to allocate container prop.");
+			return ENOMEM;
+		}
+		if (attrp->da_props != NULL) {
+			rc = daos_prop_copy(prop, attrp->da_props);
+			if (rc) {
+				daos_prop_free(prop);
+				D_ERROR("failed to copy properties "DF_RC"\n", DP_RC(rc));
+				return daos_der2errno(rc);
+			}
+		}
+		prop->dpp_entries[prop->dpp_nr - 1].dpe_type = DAOS_PROP_CO_LAYOUT_TYPE;
+		prop->dpp_entries[prop->dpp_nr - 1].dpe_val = attrp->da_type;
+		if (!uuid_is_null(attrp->da_cuuid))
+			rc = daos_cont_create(poh, attrp->da_cuuid, prop, NULL);
+		else
+			rc = daos_cont_create(poh, &attrp->da_cuuid, prop, NULL);
```

Comment:
makes sense. but let me take this in a different PR when i add a dfs_cont_create_with_label so i can do it for both dfs and generic type containers rather than going and setting the property manually here.